### PR TITLE
plane repos implementation

### PR DESCRIPTION
**Pull request by derrick-shema**

### CODE REVIEW

**Comment by derrick-shema** [comment]
[Nest] 320  - 09/10/2023, 7:50:07 AM     LOG [InstanceLoader] MongooseModule dependencies initialized +42ms
[Nest] 320  - 09/10/2023, 7:50:07 AM     LOG [InstanceLoader] MongoAirPlaneModule dependencies initialized +0ms
[Nest] 320  - 09/10/2023, 7:50:07 AM   ERROR [ExceptionHandler] Nest can't resolve dependencies of the MongoAirplaneRepository (?). Please make sure that the argument modelModel at index [0] is available in the AirplaneModule context.

Potential solutions:
- Is AirplaneModule a valid NestJS module?
- If modelModel is a provider, is it part of the current AirplaneModule?
- If modelModel is exported from a separate @Module, is that module imported within AirplaneModule?
  @Module({
    imports: [ /* the Module containing modelModel */ ]
  })

**Comment by derrick-shema** [comment]
I think the error (Nest can't resolve dependencies of the MongoAirplaneRepository (?). Please make sure that the argument modelModel at index [0] is available in the AirplaneModule context.) is resolved. Just having problems with data validation before saving in database. Tested with postman and getting 500 http code. Thinking that the issue might be in how the RegisterPlaneUseCase function parses the data or maybe in the dtos.

File: src/domain/airplane/aggregates/airplane.ts [hunk 1]
```diff
@@ -7,13 +7,13 @@ import Capacity from "../value-objects/capacity";
...
-    return this.tailNumber;
+    return this.tailNumber.getValue();
   }
   getPlaneIATACode(){
-    return this.planeIATAcode;
+    return this.planeIATAcode.getValue();
   }
   getMaxCapacity(){
-    return this.maxCapacity;
+    return this.maxCapacity.getValue();
```
Comment (kwiremo) [inline src/domain/airplane/aggregates/airplane.ts:16]: Where are you setting this values? They should be being set in the constructor

File: src/application/plane/register-plane.usecase.ts [hunk 2]
```diff
@@ -2,41 +2,41 @@ import { Injectable } from "@nestjs/common";
...
         section.seats.map(s => Seat.create(
           SeatId.createUnique(),
           SeatDesignation.create('Seat Designation',s.seatDesignation),
-          SeatStatus.AVAILABLE
+          s.status
         ))
       ))
       )
-
+      console.log(dto.cabinSections);
```
Comment (kwiremo) [inline src/application/plane/register-plane.usecase.ts]: you might not be needing these logs anymore

File: src/domain/airplane/aggregates/cabin-section.ts [hunk 3]
```diff
@@ -3,9 +3,21 @@ import CabinSectionId from "../value-objects/cabin-section-id";
...
+  getArrangement() {
+    return this.arrangement;
+  }
+  getSeats() {
+    return this.seats;
+  }
   private constructor(
-    private sectionName: string,
     cabinSectionId: CabinSectionId,
+    private sectionName: string,
```
Comment (kwiremo) [inline src/domain/airplane/aggregates/cabin-section.ts:20]: add setters

File: src/domain/airplane/entities/seat.ts [hunk 4]
```diff
@@ -4,17 +4,24 @@ import SeatStatus from "../enums/seat-status";
...
+    return this.status;
+  }
+
   constructor(
     id: SeatId,
-    seatDesignation: SeatDesignation,
-    status: SeatStatus.AVAILABLE
+    private seatDesignation: SeatDesignation,
+    private status: string
   ){
```
Comment (kwiremo) [inline src/domain/airplane/entities/seat.ts:18]: setters in constructor

File: src/domain/airplane/repositories/airplane.repository.ts [hunk 5]
```diff
@@ -3,20 +3,20 @@ import PlaneId from "../value-objects/plane-id";
...
+  save(airplane: Airplane): any;
 
   // Find an airplane by its ID
-  findById(planeId: PlaneId): Promise<Airplane | null>;
+  // findById(planeId: PlaneId): Promise<Airplane | null>;
 
-  // Find all airplanes
-  findAll(): Promise<Airplane[]>;
+  // // Find all airplanes
+  // findAll(): Promise<Airplane[]>;
```
Comment (kwiremo) [inline src/domain/airplane/repositories/airplane.repository.ts]: you can uncomment these now. It's generally not a good idea to check in commented out codes

File: src/domain/airplane/value-objects/capacity.ts [hunk 6]
```diff
@@ -2,6 +2,9 @@ import ValueObject from "src/domain/common/models/value-object";
 import ConstrainedType from "src/domain/common/primitives/constrained-type";
 
 class Capacity extends ValueObject {
+  getValue() {
+    return this.value as Readonly<number>;
+  }
   private constructor(private value: number) {
     super();
```
Comment (kwiremo) [inline src/domain/airplane/value-objects/capacity.ts:9]: setters

File: src/infrastructure/airplane/mongo-airplane.module.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { Module } from "@nestjs/common";
+import { MongooseModule } from "@nestjs/mongoose";
+import { MongoAirplaneRepository } from "./mongo-airplane.repository";
+import { AirplaneSchema } from "./airplane-schema";
+import { RegisterPlaneUseCase } from "src/application/plane/register-plane.usecase";
+
+// @Module({
+//   imports: [MongooseModule.forFeature([{ name: 'AirPlane', schema: AirplaneSchema }])],
```
Comment (kwiremo) [inline src/infrastructure/airplane/mongo-airplane.module.ts]: uncomment or remove comments

File: src/infrastructure/airplane/mongo-airplane.repository.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    }));
+
+    const airPlaneData = {
+      planeId,
+      tailNumber,
+      planeIATACode,
+      maxCapacity,
+      cabinSections
+    }
+    console.log(airPlaneData);
```
Comment (kwiremo) [inline src/infrastructure/airplane/mongo-airplane.repository.ts]: you might want to remove these consoles

**Review by kwiremo — APPROVED** [review]

**Pull request merged**